File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n) {
    vector<int> answer;
    
    int x = n;
    answer.push_back(x);
    while (x != 1) {
        if (x % 2 == 0) {
            x /= 2;
            answer.push_back(x);
        }
        
        else {
            x = 3 * x + 1;
            answer.push_back(x);
        }
    }
    
    return answer;
}

File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.py
def solution(n):
    answer = []
    
    
    x = n
    answer.append(x)
    
    while (x != 1):
        if (x % 2 == 0):
            x /= 2
            answer.append(x)
        
        else:
            x = 3 * x + 1
            answer.append(x)
    
    return answer